core/src/test/java/com/coderanch/blackjack/CardsTest.java

```diff
@@ -123,4 +123,39 @@ public void getStandardDeck_returnsUnmodifiableSet() {
     public void getStandardDeck_returnsAllCombinationsOfRankAndSuit() {
         assertThat(Cards.getStandardDeck().size(), is(equalTo(Rank.values().length * Suit.values().length)));
     }
+
+    /**
+     * Tests that {@link Cards#getShuffledStandardDeck(Random)} returns a set that contains all possible distinct cards that can be
+     * made using a combination of a {@link Rank} and a {@link Suit}.
+     */
+    @Theory
+    public void getShuffledStandardDeck_returnsAllCombinationsOfRankAndSuit() {
+        var random = new FixedRandom(List.of(5,4,2,1,3));
+        assertThat(
+            "Standard deck must contain expected number of distinct cards.",
+            Cards.getShuffledStandardDeck(random).size(),
+            is(equalTo(Rank.values().length * Suit.values().length))
+        );
+    }
+
+    private static class FixedRandom extends Random {
+
+        private List<Integer> numbers;
+        private int layer = 1;
+        private int place = 0;
+
+        public FixedRandom(List<Integer> numbers) {
+            this.numbers = numbers;
+        }
+
+        @Override
+        public int nextInt(int bound) {
+            var next = numbers.get(place++) * layer;
+            if(place > numbers.size() - 1){
```

Comment:
Formatting: please give to formatting the same importance as you give to design or anything else. That not only makes code look neat, but also who look at the source code can perceive that professionals been at work.

Formating issues:
1. After `if` statement there is no space, right before the opening parentheses.
2. There is no space after closing parentheses, right before opening curly brace.

Reply:
I normally don't block PRs because of formatting as people tend to fix. However, set auto formatter in your IDE so it complies to standard formatting rules. That way everyone saves time which could be spent on other important aspects of PR.

@nibsi what's your opinion, do you use some formatting tool so it gets reformatted by the build tool? Or you ask developers to take care of their code?

Reply:
@liutaurasvilda I just created a pull request that adds CheckStyle to the Maven build.

Reply:
sorry, i started to do reformat code in my ide.  I forgot to do it in this pr.